writer.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+const _ = require('lodash')
+const prettyOutput = require('prettyoutput')
+
+const internals = {}
+
+/**
+ * Log with pretty output formater
+ * @param {LogWriter} output
+ */
+exports.pretty = output => {
+    const log = _.assign({}, output.meta)
+    output.meta.time = output.meta.time.toISOString()
+    log.data = output.data
+    const prefix = `${output.message}\n`
+    const result = `${prefix}${prettyOutput(log, { maxDepth: 6 }, 2)}`
+    process.stdout.write(result)
+    process.stdout.write('\n')
+}
+
+/**
+ * Log in json
+ * @param {LogWriter} output
+ */
+exports.json = output => {
+    const log = _.assign({}, output.meta)
+    log.time = log.time.toISOString()
+    log.data = output.data
+    log.message = output.message
+
+    const backup = Error.prototype.toJSON
+    Error.prototype.toJSON = internals.errorToJson
```

Comment:
We should add a guard here, something like:

```javascript
if (!('toJSON' in Error.prototype)) {
```

avoiding modifying the prototype would be ideal though.

Reply:
Yes we are altering it for each log to ensure we have the proper output of errors. If not we would have to permanently alter it which could bother people.